packages/ts/react-crud/test/FormController.ts

```diff
@@ -9,39 +9,41 @@ export type FormElement = HTMLElement & {
 };
 
 export default class FormController {
-  readonly instance: FormLayoutElement;
+  readonly instance: HTMLElement;
+  readonly formLayout: FormLayoutElement;
   readonly #user: ReturnType<(typeof userEvent)['setup']>;
   readonly renderResult: HTMLElement;
 
   static async init(user: ReturnType<(typeof userEvent)['setup']>, source = document.body): Promise<FormController> {
-    const form = await waitFor(() => source.querySelector('vaadin-form-layout')!);
+    const form = await waitFor(() => within(source).getByTestId('auto-form'));
```

Comment:
This was not meant to be blocker for merging, feel free to ignore/resolve if there aren't any reasonably easy to implement solutions right now.

Reply:
This code seems to work for me:
```typescript
  readonly instance: HTMLElement;
  readonly formLayout: FormLayoutElement;
  readonly #user: ReturnType<(typeof userEvent)['setup']>;

  static async init(user: ReturnType<(typeof userEvent)['setup']>, source = document.body): Promise<FormController> {
    return Promise.resolve(new FormController(source.firstChild as HTMLElement, user));
  }

  private constructor(instance: HTMLElement, user: ReturnType<(typeof userEvent)['setup']>) {
    this.instance = instance;
    this.formLayout = instance.querySelector('vaadin-form-layout')!;
    this.#user = user;
  }
```

`renderResult` can be removed and replaced with `instance` when it's used in `autoform.spec.tsx`. Then, if this works for you, more refactoring is needed as the `init` no longer needs to be async. I kept it async to try this snippet easily.

Reply:
My main intention was to change `FormController` into something more meaningful in that it really only "controls" / interacts with the DOM that is rendered by the auto form component. Currently it just interacts with the whole document. Picking the first body element is rather random as well, and doesn't work with the crud mobile overlay tests.

If there are strong objections over adding the test ID, I'd rather keep the current approach of querying the whole document, or within `source` to have some sort of scope, and then dealing with possible problems from that later. At least at the moment it seems unlikely to cause issues.

Reply:
Actually my code snippet is not picking the first body element. It looks like that because `source` defaults to `document.body`, but in reality `source` is passed when using `init` and is the rendered element.
The code snippet above is not correctly refactored indeed: it's the most compatible with current code.
As for the overlay test, it uses another controller, I haven't looked at that one.
Putting a test ID in `src` is not like putting it on some test component in my opinion.

Reply:
To recap our discussion, there are two possible approaches:

1. use test IDs to find grids and forms, which leads to the simplest code, although it puts those IDs in the `src`;
2. rely on implementation details, like how the form is included in a crud, to find them, which leads to implementation-dependent test code, but removes test IDs.

I still have a slight preference for adding a selector which depends on our implementation. That would fail in case of implementation changes, but that's fine IMHO. But that's not a strong preference, so I'm also OK using test IDs.

Reply:
I also have no strong replacement suggestions for now, so I think it's good to be merged as it is.

Reply:
@cromoteca I see that you still have a pending review icon next to your name, so I didn't merge this. Fill free to hit the merge if there are no further reviews are planned to be done.